init.lua

```diff
@@ -177,53 +177,64 @@ function ranks.update_nametag(player)
 	end
 end
 
--- [function] Set player rank
+-- [function] Set player rank TODO
 function ranks.set_rank(player, rank)
+	local name = player
+
 	if type(player) == "string" then
 		player = minetest.get_player_by_name(player)
 	end
 
 	if registered[rank] then
+
+		storage:set_string(name, rank)
+
 		-- Set attribute
-		player:set_attribute("ranks:rank", rank)
-		-- Update nametag
-		ranks.update_nametag(player)
-		-- Update privileges
-		ranks.update_privs(player)
+		if minetest.get_player_by_name(name) then
+			player:set_attribute("ranks:rank", rank)
+			-- Update nametag
+			ranks.update_nametag(player)
+			-- Update privileges
+			ranks.update_privs(player)
+		end
 
 		return true
 	end
 end
 
 -- [function] Remove rank from player
 function ranks.remove_rank(player)
+	local name = player
+
 	if type(player) == "string" then
 		player = minetest.get_player_by_name(player)
 	end
 
-	local rank = ranks.get_rank(player)
-	if rank then
-		local name = player:get_player_name()
-
-		-- Clear attribute
-		player:set_attribute("ranks:rank", nil)
-		-- Update nametag
-		player:set_nametag_attributes({
-			text = name,
-			color = "#ffffff",
-		})
-		-- Update privileges
-		local basic_privs =
-			minetest.string_to_privs(minetest.settings:get("basic_privs") or "interact,shout")
-		minetest.set_player_privs(name, basic_privs)
+	local rank = storage:get_string(name)
+	if rank ~= "" then
```

Comment:
`ranks.get_rank` may be utilized here and we may check if it is just `nil` via `if rank then` rather than having to check if `rank` is an empty string since `ranks.get` always returns either the name of the rank or `nil`.